classes/mock/generator.php

```diff
@@ -721,6 +721,7 @@ protected function getParametersSignature(\reflectionMethod $method, $forceMockC
                     $parameterCode .= ' = ' . var_export($parameter->getDefaultValue(), true);
                     break;
 
+                case $parameter->allowsNull():
```

Comment:
There is indeed no way to directly know if a parameter is nullable (`?`) or if it is `null` by default but by using some tricks you can get the correct information:

```php
<?php

class Foo
{
    public function bar(?int $a, int $b = null) {}
}

$method = new ReflectionMethod(Foo::class, 'bar');
$parameters = $method->getParameters();

function isNullable(ReflectionParameter $parameter) {
    return $parameter->allowsNull() && !$parameter->isDefaultValueAvailable();
}

function isDefaultNull(ReflectionParameter $parameter) {
    return $parameter->allowsNull() && $parameter->isDefaultValueAvailable();
}

var_dump(
    isNullable($parameters[0]), // bool(true)
    isDefaultNull($parameters[0]), // bool(false)
    isNullable($parameters[1]), // bool(false)
    isDefaultNull($parameters[1]) // bool(true)
);
```